jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java

```diff
@@ -166,16 +166,42 @@ public final void setMaxMultiplex(int maxMultiplex)
         this.maxMultiplex = maxMultiplex;
     }
 
+    /**
+     * Get the maximum number of times the entries of the pool
+     * can be acquired.
+     * @return the max usage count.
+     */
     public int getMaxUsageCount()
     {
         return maxUsageCount;
     }
 
+    /**
+     * Change the max usage count of the pool's entries. All existing
+     * idle entries over this new max usage are removed and closed.
+     * @param maxUsageCount the max usage count.
+     */
     public final void setMaxUsageCount(int maxUsageCount)
     {
         if (maxUsageCount == 0)
             throw new IllegalArgumentException("Max usage count must be != 0");
         this.maxUsageCount = maxUsageCount;
+
+        if (closed)
+            return;
+
+        List<Entry> copy;
+        try (Locker.Lock l = locker.lock())
+        {
+            copy = new ArrayList<>(entries);
+        }
+
+        // Iterate the copy and close overused entries.
+        for (Entry entry : copy)
+        {
+            if (entry.isIdleAndOverUsed() && remove(entry) && entry.pooled instanceof Closeable)
+                IO.close((Closeable)entry.pooled);
+        }
```

Comment:
Would it not be better to do all the removals whilst the lock is held and only do the closing after releasing the lock:

```suggestion
        List<Entry> overused = new ArrayList();
        try (Locker.Lock l = locker.lock())
        {
            // look for overused here and if found remove from entries and add to overused
        }

        // Iterate the copy and close overused entries.
        for (Entry entry : overused)
        {
            if (entry.pooled instanceof Closeable)
                IO.close((Closeable)entry.pooled);
        }
```

Reply:
Good idea, that can even be improved a bit further. Let me get back to this.